README.md

```diff
@@ -7,6 +7,33 @@ A Rust driver for the Arm [PL011 UART](https://developer.arm.com/documentation/d
 
 This is not an officially supported Google product.
 
+## Usage
+
+Basic usage of the pl011-uart crate
+
+```rust
+use pl011_uart::Uart;
+
+fn main() {
+    // constants required for initializing the UART.
+    const PL011_BASE_ADDRESS: *mut u32 = 0x0900_0000 as _;
+    const PL011_BAUD_RATE: u32 = 115200;
+    const PL011_CLK_IN_HZ: u32 = 1;
+
+    // initialize PL011 UART.
+    let mut uart = unsafe { Uart::new(PL011_BASE_ADDRESS) };
+    uart.init(PL011_CLK_IN_HZ, PL011_BAUD_RATE);
+
+    // read a byte if available.
+    if let Some(byte) = uart.read_byte().unwrap() {
+        println!("reading uart: {}", byte);
+    }
+
+    // write a byte.
+    uart.write_byte(0x1F);
```

Comment:
Maybe it would be more useful to demonstrate using the `writeln!` macro, rather than just a single character.